trade-finance-manager-ui/server/controllers/login/loginService.js

```diff
@@ -0,0 +1,68 @@
+const { SSO } = require('../../constants');
+
+/**
+ * verifyReferrerForExternalSsoPost
+ * Verify referrer, allow empty referrer for localhost.
+ * 1) If host is localhost then referrer will be missing, no need to check it
+ * 2) Return error message if referrer is missing or it is not from SSO authority
+ * @param {Express.Request} req Express request
+ * @returns {boolean|string} False or error message.
+ */
+const verifyReferrerForExternalSsoPost = (req) => {
+  const hostnameWithoutPort = req.get('host').split(':')[0];
+  const referrer = req.get('Referrer');
+
+  // Referrer is not available in localhost because of policy "no-referrer-when-downgrade".
+  if (hostnameWithoutPort === 'localhost') {
+    return;
+  }
+
+  if (referrer && referrer.indexOf(`${SSO.AUTHORITY}/`) === 0) {
+    return;
+  }
+
+  console.error('Login request comming from unexpected website: %s', referrer);
```

Comment:
@abhi-markan / @ttbarnes this should already be handled by our existing and more robust csrf right?

Reply:
I think this is checking the referrer from the external Microsoft redirect so I believe needs to be checked separately from CSRF, I maybe wrong!